packages/@aws-cdk/core/test/test.logical-id.ts

```diff
@@ -254,7 +254,22 @@ export = {
       }
     });
     test.done();
-  }
+  },
+
+  'detects duplicate logical IDs in the same Stack caused by overrideLogicalId'(test: Test) {
+    const stack = new Stack();
+    const resource1 = new CfnResource(stack, 'A', { type: 'Type::Of::A' });
+    const resource2 = new CfnResource(stack, 'B', { type: 'Type::Of::B' });
+
+    resource1.overrideLogicalId('C');
+    resource2.overrideLogicalId('C');
+
+    test.throws(() => {
+      toCloudFormation(stack);
+    }, /section 'Resources' already contains 'C'/);
```

Comment:
@rix0rrr I looked into this, but it's pretty hard to do :/.

We call [`_toCloudFormation()` on each `CfnElement`](https://github.com/aws/aws-cdk/blob/1423c53fec4172ba21946ca6d33f63fc7a9d8337/packages/%40aws-cdk/core/lib/stack.ts#L834-L839) to get the CFN fragments, and we merge only those fragments. To move back from a template fragment to the `CfnElement` that generated it would be tricky - you have to have some mapping where the fragments are the key, and the `CfnElement` that generated it is the value... I guess we could use TypeScript's `Map` there maybe, but I'm not sure it's worth it...

Thoughts?

Reply:
Okay let's implement when users ask for it. 

I have plans to simplify a whole bunch but there's some effort involved that needs justification.